specifyweb/frontend/js_src/lib/components/QueryBuilder/Line.tsx

```diff
@@ -237,14 +239,20 @@ export function QueryLine({
   const hasAny = field.filters.some(({ type }) => type === 'any');
 
   const fieldSpec = React.useMemo(
-    () => QueryFieldSpec.fromPath(baseTableName, field.mappingPath),
+    () =>
+      mappingPathIsComplete(field.mappingPath)
+        ? QueryFieldSpec.fromPath(baseTableName, field.mappingPath)
+        : undefined,
     [baseTableName, field.mappingPath]
   );
 
-  const rowTableName = generateMappingPathPreview(
-    fieldSpec.baseTable.name,
-    fieldSpec.toMappingPath()
-  );
+  const rowTableName =
+    fieldSpec === undefined
+      ? fieldName
+      : generateMappingPathPreview(
```

Comment:
please move this into a React.useMemo
remember that there can be many query lines rendered at once, but the user can only modify one query line at a time, thus all others will be re-rendered uselessly and re-run generateMappingPathPreview - which is a somewhat expensive function